modules/acquia_cms_search/acquia_cms_search.module

```diff
@@ -78,6 +79,20 @@ function acquia_cms_search_field_config_insert(FieldConfigInterface $field_confi
   }
 }
 
+/**
+ * Implements hook_entity_insert().
+ */
+function acquia_cms_search_entity_insert() {
+  // Normally, content is indexed immediately after it is created or modified,
+  // at the end of the current request. But that means content created
+  // programmatically (i.e., in the PHPUnit tests) are not being indexed. So,
+  // explicitly invoke the indexer whenever an entity is created.
+  $indexer = Drupal::service('search_api.post_request_indexing');
+  if ($indexer instanceof DestructableInterface) {
+    $indexer->destruct();
+  }
+}
+
```

Comment:
This will cause issue in production site, hence we had remove this code in one of the PR, any reason to put this back?

Reply:
@chandan-singh7929 Correct. This shouldn't be there on production site.